internal/controller/kyma/controller.go

```diff
@@ -115,62 +114,45 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{RequeueAfter: r.RequeueIntervals.Success}, nil
 	}
 
-	ctx, err := r.getSyncedContext(ctx, kyma)
-
-	if !kyma.DeletionTimestamp.IsZero() && errors.Is(err, remote.ErrAccessSecretNotFound) {
-		logger.Info("access secret not found for kyma, assuming already deleted cluster")
-		r.Metrics.CleanupMetrics(kyma.Name)
-		r.removeAllFinalizers(kyma)
-
-		if err := r.updateKyma(ctx, kyma); err != nil {
-			r.Metrics.RecordRequeueReason(metrics.KymaUnderDeletionAndAccessSecretNotFound, queue.UnexpectedRequeue)
-			return ctrl.Result{}, err
+	if r.SyncKymaEnabled(kyma) {
+		err := r.SkrContextFactory.Init(ctx, kyma.GetNamespacedName())
+		if !kyma.DeletionTimestamp.IsZero() && errors.Is(err, remote.ErrAccessSecretNotFound) {
+			logger.Info("access secret not found for kyma, assuming already deleted cluster")
+			r.Metrics.CleanupMetrics(kyma.Name)
+			r.removeAllFinalizers(kyma)
+
+			if err := r.updateKyma(ctx, kyma); err != nil {
+				r.Metrics.RecordRequeueReason(metrics.KymaUnderDeletionAndAccessSecretNotFound, queue.UnexpectedRequeue)
+				return ctrl.Result{}, err
+			}
+			r.Metrics.RecordRequeueReason(metrics.KymaUnderDeletionAndAccessSecretNotFound, queue.IntendedRequeue)
+			return ctrl.Result{Requeue: true}, nil
 		}
-		r.Metrics.RecordRequeueReason(metrics.KymaUnderDeletionAndAccessSecretNotFound, queue.IntendedRequeue)
-		return ctrl.Result{Requeue: true}, nil
-	}
 
-	// Prevent ETCD load bursts during secret rotation
-	if apierrors.IsUnauthorized(err) {
-		r.deleteRemoteClientCache(ctx, kyma)
-		r.Metrics.RecordRequeueReason(metrics.KymaUnauthorized, queue.UnexpectedRequeue)
-		return r.requeueWithError(ctx, kyma, err)
-	}
+		// Prevent ETCD load bursts during secret rotation
+		if apierrors.IsUnauthorized(err) {
+			r.deleteRemoteClientCache(ctx, kyma)
+			r.Metrics.RecordRequeueReason(metrics.KymaUnauthorized, queue.UnexpectedRequeue)
+			return r.requeueWithError(ctx, kyma, err)
+		}
 
-	if err != nil {
-		r.deleteRemoteClientCache(ctx, kyma)
-		r.Metrics.RecordRequeueReason(metrics.SyncContextRetrieval, queue.UnexpectedRequeue)
-		return r.requeueWithError(ctx, kyma, err)
+		if err != nil {
+			r.deleteRemoteClientCache(ctx, kyma)
+			r.Metrics.RecordRequeueReason(metrics.SyncContextRetrieval, queue.UnexpectedRequeue)
+			return r.requeueWithError(ctx, kyma, err)
+		}
 	}
 
 	return r.reconcile(ctx, kyma)
 }
 
-func (r *KymaReconciler) deleteRemoteClientCache(ctx context.Context, kyma *v1beta2.Kyma) {
+func (r *Reconciler) deleteRemoteClientCache(ctx context.Context, kyma *v1beta2.Kyma) {
 	logger := logf.FromContext(ctx)
 	logger.Info("connection refused, assuming connection is invalid and resetting cache-entry for kyma")
 	r.RemoteClientCache.Delete(client.ObjectKeyFromObject(kyma))
```

Comment:
you can also use `kyma.GetNamespacedName()`, right?

Reply:
check this comment for further improvement https://github.com/kyma-project/lifecycle-manager/pull/1426/files#r1594129327